service/worker/replicator/replicationTask.go

```diff
@@ -225,30 +301,44 @@ func (t *activityReplicationTask) HandleErr(err error) error {
 		return err
 	}
 
-	retryErr, ok := t.convertRetryTaskError(err)
-	if !ok || retryErr.GetRunId() == "" {
+	retryV1Err, okV1 := t.convertRetryTaskError(err)
+	_, okV2 := t.convertRetryTaskV2Error(err)
```

Comment:
shouldn't this be called  only when okv1 is false? In that case you don't need to put okv1 and okv2 together

Reply:
forgot to remove 
```go
if !okV1 || retryV1Err.GetRunId() == "" {
    return err
}
```